modular_skyrat/modules/implants/code/augments_chest.dm

```diff
@@ -6,14 +6,19 @@
 	icon_state = "internal_HA"
 	implant_overlay = null
 	implant_color = null
-	actions_types = list(/datum/action/item_action/organ_action/use)
+	actions_types = list(/datum/action/item_action/organ_action/use/internal_analyzer)
 	w_class = WEIGHT_CLASS_SMALL
 
-/obj/item/organ/internal/cyberimp/chest/scanner/ui_action_click(owner, action)
-	if(istype(action, /datum/action/item_action/organ_action/use))
-		if(organ_flags & ORGAN_FAILING)
-			to_chat(owner, span_warning("Your health analyzer relays an error! It can't interface with your body in its current condition!"))
-			return
-		else
-			healthscan(owner, owner, 1, TRUE)
-			chemscan(owner, owner)
+/datum/action/item_action/organ_action/use/internal_analyzer
+	desc = "LMB: Health scan. RMB: Chemical scan. Requires implanted analyzer to not be failing due to EMPs or other causes. Does not provide treatment assistance."
+
+/datum/action/item_action/organ_action/use/internal_analyzer/Trigger(trigger_flags)
+	. = ..()
+	var/obj/item/organ/internal/cyberimp/chest/scanner/our_scanner = target
+	if(our_scanner.organ_flags & ORGAN_FAILING)
+		to_chat(owner, span_warning("Your health analyzer relays an error! It can't interface with your body in its current condition!"))
+		return
+	if(trigger_flags & TRIGGER_SECONDARY_ACTION)
+		chemscan(owner, owner)
+	else
+		healthscan(owner, owner, 1, TRUE)
```

Comment:
```suggestion
		healthscan(owner, owner, SCANNER_VERBOSE, TRUE)
```

Reply:
as discussed in dev channel, can't do this; not defined here, it was defined and then undefined in the health scanner file